src/EVM/Expr.hs

```diff
@@ -1232,6 +1234,9 @@ simplifyProp prop =
     go (PNeg (PBool b)) = PBool (Prelude.not b)
     go (PNeg (PNeg a)) = a
 
+    -- Empty buf
+    go (PEq (Lit 0) (BufLength k)) = peq k (ConcreteBuf "")
```

Comment:
Are you sure this is a simplification?
Is it better on the SMT level to compare arrays instead of computing buffer length and comparing it to `0`?